dvclive/data/plot.py

```diff
@@ -46,8 +46,36 @@ def step_dump(self) -> None:
             "DVCLive plots can only be used in no-step mode."
         )
 
+    @classmethod
+    def get_template(cls):
+        from dvc_render.vega_templates import LinearTemplate
+
+        return LinearTemplate()
+
+    @classmethod
+    def get_properties(cls):
+        raise NotImplementedError
+
+    @classmethod
+    def dump_template(cls, output_file):
+        template = cls.get_template()
+        content = template.content
+        for k, v in cls.get_properties().items():
+            content = template.fill_anchor(content, k, v)
+        Path(output_file).write_text(content)
```

Comment:
All this logic is to somehow allow preserving behavior with and without DVC by serializing templates. So users can set in DVC.yaml:

```
stages:
  test-dvclive:
    cmd: python test_dvclive.py
    plots:
      - dvclive/plots/confusion_matrix.json:
          template: dvclive/plots/confusion_matrix.template
```

DVCLive would be able to work without the serialized template.

Nor sure if the templates should be dumped to a separate subfolder (i.e. `dvclive/templates` instead of `dvclive/plots`) in order to be preserved across runs:

```
stages:
  test-dvclive:
    cmd: python test_dvclive.py
    plots:
      - dvclive/plots/confusion_matrix.json:
          template: dvclive/templates/confusion_matrix.template
```

Reply:
Is this only necessary for `dvc plots`? Is serialization needed for dvclive to make the report?

Reply:
> Is this only necessary for dvc plots?

Yes

> Is serialization needed for dvclive to make the report?

It's not needed, DVCLive could generate the report without dumping templates.

Reply:
IMO it might be better to at least put this in a follow-up PR because it might need more discussion or a different approach might come up. Even without it, we are achieving parity with current DVC behavior, right?

Reply:
> IMO it might be better to at least put this in a follow-up PR because it might need more discussion or a different approach might come up. Even without it, we are achieving parity with current DVC behavior, right?

Removed from this P.R.